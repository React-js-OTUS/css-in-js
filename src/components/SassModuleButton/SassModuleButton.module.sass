@import src/variables

.root
  :global(.some-global-class)
    color: $pink

.button
  border-radius: 8px
  font-size: 18px
  padding: 8px 16px
  outline: none
  border: 0
  background-color: $orange
  color: #fff
  cursor: pointer

  &:hover
    background-color: $orangeLight

  &:active
    background-color: $orangeDark

.text
  +little-text()

  &.opened
    display: block

  &.closed
    display: none
