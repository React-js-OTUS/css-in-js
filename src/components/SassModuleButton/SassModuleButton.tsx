import React, {FC, useState} from 'react';
import cn from 'clsx';
import variables from '../../variables.scss';
import s from './SassModuleButton.module.sass';

export type ButtonProps = React.HTMLAttributes<HTMLButtonElement> & {
  className?: string;
};

export const SassModuleButton: FC<ButtonProps> = ({ className, ...props }) => {
  const [opened, setOpened] = useState<boolean>(false);
  return (
    <div className={cn(s.root, className)}>
      <button className={s.button} {...props} onClick={() => setOpened(v => !v)} />
      <div className={cn(s.text, opened ? s.opened : s.closed)}>Test</div>
      <div className={cn(s.text, !opened && s.closed)}>Test</div>
      <div className="some-global-class">Global</div>
      <div style={{ color: variables.orange }}>Import Variable</div>
    </div>
  );
}