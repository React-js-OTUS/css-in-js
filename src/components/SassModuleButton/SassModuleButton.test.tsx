import React from 'react';
import { render } from '@testing-library/react';
import { SassModuleButton } from './SassModuleButton';

describe('SassModuleButton', () => {
  it('base snapshot', () => {
    const { container } = render(<SassModuleButton>SassModuleButton</SassModuleButton>);
    expect(container).toMatchSnapshot();
  });
});
