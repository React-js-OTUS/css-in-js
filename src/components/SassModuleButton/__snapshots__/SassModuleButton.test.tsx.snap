// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SassModuleButton base snapshot 1`] = `
<div>
  <div
    class="root"
  >
    <button
      class="button"
    >
      SassModuleButton
    </button>
    <div
      class="text closed"
    >
      Test
    </div>
    <div
      class="text closed"
    >
      Test
    </div>
    <div
      class="some-global-class"
    >
      Global
    </div>
    <div
      style="color: orange;"
    >
      Import Variable
    </div>
  </div>
</div>
`;
